File: baitap-submit/huynh_thien_ba/02-llm-api-params/final.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('Enter the coefficient of x^3: ', (a) => {
  rl.question('Enter the coefficient of x^2: ', (b) => {
    rl.question('Enter the coefficient of x: ', (c) => {
      rl.question('Enter the constant term: ', (d) => {
        try {
          const roots = solveCubicEquation(parseFloat(a), parseFloat(b), parseFloat(c), parseFloat(d));
          console.log('Roots of the equation are:', roots);
        } catch (error) {
          console.error('Invalid input. Please enter numerical values.');
        }
        rl.close();
      });
    });
  });
});

function solveCubicEquation(a, b, c, d) {
  if (a === 0) {
    // Handle the case when the equation is not cubic
    throw new Error('The coefficient of x^3 cannot be zero.');
  }

  // Implement the cubic equation solving algorithm here
  // ...
}

File: baitap-submit/huynh_thien_ba/02-llm-api-params/final.py
from cmath import sqrt

def solve_cubic_equation(a, b, c, d):
  if a == 0:
    raise ValueError("Coefficient a cannot be zero for a cubic equation")
  delta = (b**2 - 4*a*c)
  if delta >= 0:
    x1 = (-b - sqrt(delta)) / (2*a)
    x2 = (-b + sqrt(delta)) / (2*a)
    return x1, x2
  else:
    x1 = (-b - sqrt(abs(delta))) / (2*a)
    x2 = (-b + sqrt(abs(delta))) / (2*a)
    x3 = (-b) / (2*a)
    return x1, x2, x3